packages/vite/src/node/plugins/define.ts

```diff
@@ -99,6 +102,7 @@ export function definePlugin(config: ResolvedConfig): Plugin {
         // exclude html, css and static assets for performance
         isHTMLRequest(id) ||
         isCSSRequest(id) ||
+        isNonJsRequest(id) ||
         config.assetsInclude(id)
```

Comment:
I wonder if we should go the other way around and use the `isJSRequest(id)` utility here. This isn't only for performance, if we want to avoid inconsistencies between dev and build, we shouldn't be doing replacements for define in non-js sources.
I think we can merge your PR now to close the issue, but let's explore this further.

Reply:
I've thought about it too but it feels like a bold move that could cause a breaking change if there are more extensions that compile to JS (Vite currently keeps a hardcoded list for it). And I can see some cases for this, like `.svelte.md` which is compiled by Svelte but is written in markdown. 

Maybe the more robust way is to somehow identify the code is JS and then do the replacements. Or otherwise we can proceed with https://github.com/vitejs/vite/pull/3828 to make things more flexible.

Reply:
Yes, #3828 got stuck because we thought there was a chance of removing this list altogether but there hasn't been progress in that direction and this is another valid use case for being able to answer the "isJSRequest" question. Maybe we should push for #3828 again.